createtransfers/createtransfers.py

```diff
@@ -194,17 +199,137 @@ def create_variously_encoded_files():
                 return
 
 
-def create_large_test_transfers():
+def create_variously_encoded_dir_names(zip_path=None):
+    """Create folders with strange (non-UTF8) encodings under
+    TestTransfers/dirs_with_various_encodings/
+    """
+    sub_dir = 'dirs_with_various_encodings'
+    target_path = os.path.join(HERE, 'TestTransfers', sub_dir)
+    if zip_path:
+        target_path = os.path.join(zip_path)
+    LOGGER.info("Transfer target path: %s", target_path)
+    for encoding_info in VARIOUS_ENCODINGS:
+        # Use the filename from the VARIOUS_ENCODINGS tuples minux the file
+        # extension to give us something that looks like a filename.
+        dir_names = encoding_info['file_name']
+        if not isinstance(dir_names, (tuple, list)):
+            dir_names = (dir_names,)
+        for dir_ in dir_names:
+            encoding_dir_path = os.path.join(
+                target_path, encoding_info['dir_name'],
+                os.path.splitext(os.path.basename(dir_))[0])
+            try:
+                rm_dirs_and_create(encoding_dir_path)
+            except CreateTransferException as err:
+                LOGGER.error(err)
+                return
+            encoding = encoding_info['encoding']
+            try:
+                create_file_and_write("{}_encoded_dirs.txt".format(encoding),
+                                      encoding_dir_path,
+                                      encoding,
+                                      dirs=True)
+            except CreateTransferException as err:
+                LOGGER.error(err)
+                return
+
+
+def create_zip_packages():
```

Comment:
This function seems quite long. It would be nice if it could be broken up into more readable chunks, but more a nice-to-have than a substantive criticism.

Reply:
With the `MakeFile` I have now made this into four separate functions.